recipe-client-addon/lib/AboutStudiesProtocol.jsm

```diff
@@ -0,0 +1,105 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+"use strict";
+
+const {utils: Cu, interfaces: Ci, manager: Cm, results: Cr} = Components;
+Cm.QueryInterface(Ci.nsIComponentRegistrar);
+
+Cu.import("resource://gre/modules/Services.jsm");
+Cu.import("resource://gre/modules/XPCOMUtils.jsm");
+
+this.EXPORTED_SYMBOLS = ["AboutStudiesProtocol"];
+
+/**
+ * Component definition for the about:studies protocol handler.
+ * Registers a component with the browser that establishes an `about:studies`
+ * protocol handler. Navigating to `about:studies` displays the
+ * `AboutStudies.xml` file.
+ */
+function StudiesProtocolHandler() {}
+StudiesProtocolHandler.prototype = {
```

Comment:
Can this be either a class or a plain object? What's the benefit of using the prototype way of doing things?